bimsb/packages/bioinformatics-nonfree.scm

```diff
@@ -1882,3 +1883,100 @@ MACE is a bioinformatics tool dedicated to analyze ChIP-exo data.")
     ;; others say "free for all use", others say "GPL version 1.3" (?), yet
     ;; others say "Artistic license".
     (license license:gpl2+)))
+
+;; Although this program is released under the GPL it depends on
+;; ViennaRNA and bowtie1, which are non-free software.
+(define-public mirdeep2
+  (package
+    (name "mirdeep2")
+    (version "0.1.0")
+    (source (origin
+              (method git-fetch)
+              (uri (git-reference
+                   (url "https://github.com/rajewsky-lab/mirdeep2.git")
+                   (commit (string-append "v" version))))
+              (file-name (string-append name "-" version "-checkout"))
+              (sha256
+               (base32
+                "0srxxymxmpfb656g6r26zhplx9fl6wxrzzdl6wr7r06nnmx6sjd2"))))
+    (build-system gnu-build-system)
+    (arguments
+     `(#:tests? #f ;; no check target
+       #:phases
+       (modify-phases %standard-phases
+         (delete 'configure)
+         (replace 'build
+           (lambda* (#:key inputs outputs #:allow-other-keys)
+             (let ((out (assoc-ref outputs "out")))
+               ;; patch scripts checking for ../install_successful file
+               (for-each
+                 (lambda (script)
+                   (substitute* script
+                     (("\\$bn/../install_successful")
+                      (string-append out
+                                     "/share/mirdeep2/install_successful"))))
+                 '("src/mapper.pl"
+                   "src/miRDeep2.pl"
+                   "src/quantifier.pl"))
+               ;; patch script using ../Rfam_for_miRDeep.fa file
+               (substitute* "src/miRDeep2.pl"
+                 (("\\$\\{scripts\\}/../Rfam_for_miRDeep.fa")
+                  (string-append out "/share/mirdeep2/Rfam_for_miRDeep.fa"))
+                 (("\\$scripts/../Rfam_for_miRDeep.fa")
+                  (string-append out "/share/mirdeep2/Rfam_for_miRDeep.fa"))))
+               ;; patch script using $(dirname `which miRDeep2.pl`)/indexes dir
+               (substitute* "src/make_html.pl"
+                 (("`which miRDeep2.pl`") "`realpath ~/.local/share/mirdeep2`")
+                 (("mkdir \"\\$\\{scripts\\}indexes")
+                  "mkdir -p \"${scripts}indexes"))
```

Comment:
Also here I wonder if I patch 'properly', especially w.r.t. escaping. :wink:

Reply:
this looks correct to me.

Reply:
@rekado 
> Yes, sounds fine, though you'd need to make sure that this directory exists.

That's why I patch in the `-p` here to the corresponding `mkdir` call in miRDeep2 - so the directory will be created when needed as long as the user permissions allow the creation of `~/.local/share/mirdeep2`.
I figured that's safe to assume. :wink: